scene/resources/particles_material.cpp

```diff
@@ -117,10 +128,12 @@ void ParticlesMaterial::finish_shaders() {
 }
 
 void ParticlesMaterial::_update_shader() {
+	print_line("updating shaders...");
```

Comment:
Don't use `print_line` - it's good for debugging but bad for the end-users.

Reply:
damn, forgot it again... but it seems there is still some things to do...